### DISCUSSION TITLE
ping not happening automatically

### DISCUSSION THREAD

**Discussion by artknight** [uNetworking/uWebSockets.js#530] [Q&A] [answered]
As per the docs at ( https://github.com/uNetworking/uWebSockets/blob/master/misc/READMORE.md ) a ping is supposed to happen automatically. `The library will autoamtically send pings to clients acording to the idleTimeout specified`

However, I am not seeing any pings. Is there something I need to setup?

Note: I am using uWebsockets.js v19.2.0

**Comment by hst-m** [comment, marked as answer]
> I am not seeing any pings

How are you checking for pings? Browser does not show pings. You will get a Pong event on the server

**Reply by artknight** [reply]
@hst-m I added a `ping` listener to the websocket. This is on the client side

```
this.websocket.addEventListener('ping', this.onPing.bind(this));
```

**Reply by hst-m** [reply]
There is no ping event on browser, is that a browser or what? Did you check Pong event on server?

**Reply by artknight** [reply]
@hst-m Yes, I am seeing a pong event on the server!

How can I check it on the client side? Is there an event I can subscribe to?

**Reply by hst-m** [reply]
Please don't come on here and spam a bunch of posts if you have no idea what you are doing, where did you get idea there is Ping event on browser? Do you know how to do a google search? 2 posts in a row wasting peoples time

**Reply by artknight** [reply]
@hst-m Wow, that is very rude! I am guessing you are having a very bad day!

**Reply by hst-m** [reply]
I dont like wasting my time trying to help you troubleshoot because you are unable to do a google search

**Reply by artknight** [reply]
@hst-m I have spent some time trying to find answers. Unfortunately I did not find what I was looking for and thought asking this community would be the way to go. You do not need to be rude and you can simply not reply my questions. No one asked you specifically to help me!

**Reply by hst-m** [reply]
I am just confused, where did you get idea there is Ping event on browser? You pull that out of no where? Making stuff up as you go?

**Reply by artknight** [reply]
@hst-m actually I saw it when looking at this lib https://www.npmjs.com/package/ws and I thought that there might be something similar here. Sorry.

**Reply by hst-m** [reply]
That's where I figured you got it from, obviously those are different things, if you are using Browser websocket you should probably be looking at the Browser websocket API docs https://developer.mozilla.org/en-US/docs/Web/API/WebSocket

**Reply by artknight** [reply]
@hst-m right, well since that `ws` lib is able to emit ping events down to clients I naturally thought that uWebsockets.js lib could do the same. That is why I asked the community if there is a way to listen to the `pong` event in the browser. 

No reason to fly off the handle!

**Reply by hst-m** [reply]
> well since that ws lib is able to emit ping events down to clients I naturally thought that uWebsockets.js lib could do the same

I think you are very confused, both libraries send pings to client, this library can do it automatically and manually with ws.ping, but a browser client does not show pings as I tried to tell you many times and you did not listen, the ws library can work as client or server and it is able to show client pings because it is not limited like a browser

**Comment by ghost** [comment]
If you get pong event on the server then pings do work. The only way to get pong events on the server is if the client sent you a pong. The client will only send a pong if it got a ping.

**Reply by artknight** [reply]
@alexhultman what do i need to do on the client side ( browser ) in order to respond to the `pong` from the server? 

I am assuming I need to `ping` the server back?

**Reply by hst-m** [reply]
Omg you are back making zero sense again, and I already explained this to you like 3 times. Pings are sent from server, Pong is automatically replied back from the browser. You already said you saw the Pong event on the server which was sent from the Browser. This is really not that difficult

**Reply by artknight** [reply]
@hst-m dude since I am not even addressing you, please stay out of this discussion!

**Reply by hst-m** [reply]
No one wants to waste time with someone so clueless, you don't understand how basic Ping/Pong works

**Reply by artknight** [reply]
@hst-m yes i am confused because **I am not sending any pings from the browser**, unless the browser WebSocket automatically sends a ping to the server. If I understand what you are saying correctly the uWebsockets.js lib will keep the websocket open automatically without having me implement anything on the client side?

**Reply by artknight** [reply]
Btw, @hst-m you are a bully! I hope you lose your privileges to be able to snub anyone else!

**Reply by hst-m** [reply]
I don't know how many times this needs to be explained to you, in addition to the explanation in the Readme, Server automatically sends Pings to browser, Browser automatically sends Pong responses back to Server. The Browser does not send Ping to Server, it sends Pong which is a response to a Ping. Connection is then kept open. I am expecting to need to explain this at least 3 more times

**Comment by artknight** [comment]
Thank you.
  
  
  
    
  

  
  
>
